src/cid.js

```diff
@@ -331,7 +331,8 @@ export class CID {
 
     let version = next()
     let codec = DAG_PB_CODE
-    if (version === 18) { // CIDv0
+
+    if (version === 18 || version === 0) { // CIDv0 sha2-256 or identity
```

Comment:
PeerIDs are not CIDs by spec https://github.com/libp2p/specs/blob/master/peer-ids/peer-ids.md, https://github.com/multiformats/cid.

CIDv0 is always DAG-PB and sha2-256 (and encoded as base58btc).

Conflating these specs may lead to user confusion here. In go-libp2p peerIDs are defined as a separate type (i.e. https://github.com/libp2p/go-libp2p-core/blob/525a0b13017263bde889a3295fa2e4212d7af8c5/peer/peer.go#L172 vs https://github.com/ipfs/go-cid/blob/979bf3fb8572224c2b2fbfaf153f94b98734807c/cid.go#L243)

Reply:
> PeerIDs are not CIDs by spec

Hmm: https://github.com/libp2p/specs/blob/master/RFC/0001-text-peerid-cid.md

The `peer-id` module carries a lot of crypto baggage with it that makes it unsuitable for use in the http client as it's too large, to the point that we don't use it in the http client, only in in-process or daemon js-ipfs/js-libp2p.

For whatever reason we decided instead to pass peer IDs around as CIDs as a 'lighter weight' version of `peer-id` so it has methods to convert to and from a CID (v0 `identity` or v1 `libp2p-key`).

Maybe it shouldn't but that's a much bigger scope change than upgrading multiformats to get the new `@ipld/dag-cbor` `@ipld/dag-pb` module goodness.

All this PR really does is add a feature that was in the `cids` module but is missing from this one.

Reply:
> Hmm: https://github.com/libp2p/specs/blob/master/RFC/0001-text-peerid-cid.md

This is a spec detailing how to encode peerIDs as CIDs in text, which is a subset of the peerID spec (e.g. peerIDs are represented as multihashes, not CIDs, in byte encoding and peerIDs are also representable as base58btc multihashes in text).

> All this PR really does is add a feature that was in the cids module but is missing from this one.

🤷‍♀️ if the previous implementation (js-cid) wasn't spec compliant and the idea is you want to make this one match the other one then I guess up to you + others familiar with the js codebase. You should probably file an issue then to make a peer-id type that you can use so that you can make the CID type spec compliant again.